test2code/src/main/kotlin/com/epam/drill/test2code/coverage/CoverageSender.kt

```diff
@@ -15,49 +15,90 @@
  */
 package com.epam.drill.test2code.coverage
 
+import com.epam.drill.plugins.test2code.common.api.*
+import com.github.luben.zstd.*
 import kotlinx.coroutines.*
-import mu.KotlinLogging
-import java.util.concurrent.Executors
-import kotlin.coroutines.CoroutineContext
+import kotlinx.serialization.protobuf.*
+import mu.*
+import java.util.*
+import java.util.concurrent.*
+
+private const val RETENTION_QUEUE_LIMIT = 2000L
 
 interface CoverageSender {
-    fun setSendingHandler(handler: SendingHandler)
     fun startSendingCoverage()
     fun stopSendingCoverage()
 }
 
 class IntervalCoverageSender(
-    intervalMs: Long,
-    collectProbes: () -> Sequence<ExecDatum> = { emptySequence() }
+    private val intervalMs: Long,
+    private val coverageTransport: CoverageTransport,
+    private val inMemoryRetentionQueue: InMemoryRetentionQueue = InMemoryRetentionQueue(totalSizeByteLimit = RETENTION_QUEUE_LIMIT),
```

Comment:
Change val type to the `: RetentionQueue`. It is important since `InMemoryRetentionQueue` is just an implementation.
`IntervalCoverageSender` must not request exact implementation, it should request contract.
(That will be easier to unit test too)

Reply:
done